src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageType.cs

```diff
@@ -96,7 +96,7 @@ public bool Equals(MessageType other)
             {
                 return true;
             }
-            return Equals(other.TypeName, TypeName) && other.Version.Major == Version.Major;
+            return Equals(other.TypeName, TypeName);
```

Comment:
yes breaking behavior change but in this case we see this as a bug so I'd say we're good